hazelcast/src/hazelcast/client/xml_impl.cpp

```diff
@@ -0,0 +1,1317 @@
+            }
+            std::ifstream stream;
+            stream.open(resource);
+            if (stream.fail()) {
+                throw hazelcast::client::exception::invalid_configuration(
+                  "Failed to load resource: " + resource);
+            }
+            if (!currently_imported_files.emplace(resource).second) {
+                throw hazelcast::client::exception::invalid_configuration(
+                  "Resource '" + resource +
+                  "' is already loaded! This can be due to" +
+                  " duplicate or cyclic imports.");
+            }
+            boost::property_tree::ptree imported_root = parse(std::move(stream));
+            try{
+                imported_root = imported_root.get_child("hazelcast-client");
+            }catch(const boost::exception& e){
+                throw hazelcast::client::exception::invalid_configuration("Imported file " + resource + " is invalid");
+            }
+            replace_imports(&imported_root);
+            for (auto& imported_node : imported_root) {
+                if (imported_node.first == "<xmlattr>") {
+                    continue;
+                }
+                root->put_child(imported_node.first, imported_node.second);
+            }
+            /* Deletion is not necessary / deleting a member of something while iterating over it is a bad idea
+            for (auto it = root->begin(); it != root->end(); ++it) {
+                if (it->first == "import") {
+                    it = root->erase(it);
+                    break;
+                }
+            }*/
+        }
+    }
+
+
+
+}
+
+property_replacer
+abstract_xml_config_builder::create_replacer(
+  const boost::property_tree::ptree& node)
+{
+    std::string replacer_class = get_attribute(node, "class-name");
+    std::unordered_map<std::string, std::string>  properties_;
+    for (auto& n : node) {
+        std::string value = n.first;
+        if ("properties" == value) {
+            fill_properties(n.second, &properties_);
+        }
+    }
+    property_replacer replacer;
+    replacer.init(std::move(properties_));//std::move
+    return replacer;
+}
+
+std::string
+abstract_xml_config_helper::get_namespace_type()
+{
+    return "client-config";
+}
+abstract_xml_config_builder::abstract_xml_config_builder() = default;
+
+void
+abstract_xml_config_builder::fill_properties(
+  const boost::property_tree::ptree& node,
+  std::unordered_map<std::string, std::string>* properties_)
+{
+    abstract_dom_config_processor::fill_properties(
+      node, properties_, dom_level_3);
+}
+
+xml_client_config_builder::xml_client_config_builder(
+  xml_client_config_locator* locator)
+{
+    this->in = locator->get_in();
+}
+boost::property_tree::ptree
+xml_client_config_builder::parse(std::ifstream input_stream)
+{
+    boost::property_tree::ptree tree;
+    try {
+        boost::property_tree::read_xml(input_stream, tree);
+        input_stream.close();
```

Comment:
std::ifstream::close() calls should be removed, it uses RAII technique, so it calls close() at its destructor.

``` suggestion
```